functions/src/validation.ts

```diff
@@ -0,0 +1,78 @@
+import {
+  body,
+  query,
+  ValidationError,
+  validationResult,
+} from "express-validator";
+import { NextFunction, Request, Response } from "express";
+import { APIFields, isValidUrl, ShareLinksCollection } from "./utils";
+import {
+  getShareLinkErrorByCode,
+  sendAndThrowShareLinkErrorCode,
+  ShareLinkErrorCode,
+} from "./error-handling";
+import * as functions from "firebase-functions";
+
+export function registerUrlValidationRules() {
+  return [
+    // Use a custom URL validator b/c we need to ensure an HTTP(S) protocol is
+    // present for use in the /go/ endpoint.
+    body(ShareLinksCollection.URL).custom(isValidUrl),
+    body(ShareLinksCollection.IMAGE_URL).optional().custom(isValidUrl),
+    body(ShareLinksCollection.TITLE).optional().isString(),
+    body(ShareLinksCollection.DESCRIPTION).optional().isString(),
+    body(ShareLinksCollection.IMAGE_HEIGHT).optional().isInt(),
+    body(ShareLinksCollection.IMAGE_WIDTH).optional().isInt(),
```

Comment:
This validation stuff is neat. Thanks for figuring it out and cleaning up the code!